File: exercises/D503/solution.c
#include <stdio.h>
int a[6];
int main() {
    scanf("%d %d %d %d %d %d", &a[0], &a[1], &a[2], &a[3], &a[4], &a[5]);
    FILE* fout = fopen("chart.txt", "w");
    fprintf(fout, "+-------------------------+\n");
    int i, j;
    for (i = 10; i >= 1; i--) {
        fprintf(fout, "| ");
        for (j = 0; j < 6; j++) {
            if (a[j] >= i) {
                fprintf(fout, "### ");
            } else {
                fprintf(fout, "    ");
            }
        }
        fprintf(fout, "|\n");
    }
    fprintf(fout, "+-------------------------+\n");
    fclose(fout);
    return 0;
}

File: exercises/D503/solution.java
import java.util.*;
import java.io.FileWriter;
import java.io.IOException;

class D503 {
    public static void main(String[] args) {
        try {
            FileWriter fout = new FileWriter("chart.txt");
            Scanner in = new Scanner(System.in);
            int []a = new int[6];
            for (int i = 0; i < 6; i++) {
                a[i] = in.nextInt();
            }
            fout.write("+-------------------------+\n");
            for (int i = 10; i >= 1; i--) {
                fout.write("| ");
                for (int j = 0; j < 6; j++) {
                    if (a[j] >= i) {
                        fout.write("### ");
                    } else {
                        fout.write("    ");
                    }
                }
                fout.write("|\n");
            }
            fout.write("+-------------------------+\n");
            fout.close();
        } catch (IOException e) {
            System.out.println("File IO exception");
        }
        
    }
}
